perun-base/src/main/java/cz/metacentrum/perun/core/api/Group.java

```diff
@@ -120,14 +120,18 @@ public void setShortName(String shortName) {
 		}
 	}
 
-
-
-	public int compareTo(Group group) {
-		if (group == null) throw new InternalErrorRuntimeException(new NullPointerException("Group group"));
-		if (this.getId() != group.getId()) {
-			return this.name.toLowerCase().compareTo(group.getName().toLowerCase());
+	@Override
+	public int compareTo(PerunBean perunBean) {
+		if(perunBean == null) throw new NullPointerException("PerunBean to compare with is null.");
+		if(perunBean instanceof Group) {
+			Group group = (Group) perunBean;
+			if (this.getName() == null && group.getName() != null) return -1;
+			if (group.getName() == null && this.getName() != null) return 1;
+			if (this.getName() == null && group.getName() == null) return 0;
+			return this.getName().toLowerCase().compareToIgnoreCase(group.getName());
```

Comment:
Any reason to convert ``this.getName()`` to lowercase when you are comparing using ignore case ? Also this method should honor ``equals()`` when possible. => Two groups with same name but different case are different for Perun based on equals().